File: arrays/basics/second_large.ts
function secondLargest(arr: number[]) {
  let largest = arr[0];
  let secondlg = arr[0];
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > largest) {
      secondlg = largest;
      largest = arr[i];
    } else if (arr[i] > secondlg && arr[i] < largest) {
      secondlg = arr[i];
    }
  }
  return secondlg;
}
let ar1 = [10, 50, 40, 30, 20, 10];

console.log(secondLargest(ar1));

File: arrays/basics/py/second_large.py

def secondLarge(arr):
    largest=arr[0];
    secondLargest=arr[0];
    for i in range(len(arr)):
        if(arr[i]>largest):
            secondLargest=largest;
            largest=arr[i]
        elif(arr[i]>secondLargest and arr[i]<largest):
            secondLargest=arr[i]
    return secondLargest

arr=[10,20,50,30,50,30];
print(f"Second Largest : {secondLarge(arr)}")